### PR TITLE
[Messenger] Fix Redis integration tests data provider

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

This fixes the data provider for `RedisExtIntegrationTest` and add `redis_sentinel` as a valid option

### CODE REVIEW

File: src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/RedisExtIntegrationTest.php [hunk 1]
```diff
@@ -221,18 +221,14 @@ public function testConnectionClaimAndRedeliver()
...
     public function testSentinel(string $sentinelOptionName)
     {
         if (!$hosts = getenv('REDIS_SENTINEL_HOSTS')) {
             $this->markTestSkipped('REDIS_SENTINEL_HOSTS env var is not defined.');
         }
 
-        if (!getenv('MESSENGER_REDIS_SENTINEL_MASTER')) {
-            $this->markTestSkipped('MESSENGER_REDIS_SENTINEL_MASTER env var is not defined.');
-        }
-
```
Comment (alexandre-daubois) [inline src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/RedisExtIntegrationTest.php:235]: This can be removed as we fallback on null, like other tests in this file

File: src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php [hunk 2]
```diff
@@ -47,6 +47,7 @@ class Connection
         'auth' => null,
         'serializer' => 1, // see \Redis::SERIALIZER_PHP,
         'sentinel_master' => null, // String, master to look for (optional, default is NULL meaning Sentinel support is disabled)
+        'redis_sentinel' => null, // String, alias for 'sentinel_master'
```
Comment (xabbuh) [inline src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php:50]: Is this change necessary?

File: src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/RedisExtIntegrationTest.php [hunk 3]
```diff
@@ -252,10 +252,11 @@ public function testSentinel(string $sentinelOptionName)
         $connection->cleanup();
     }
 
-    public function sentinelOptionNames(): iterable
+    public static function sentinelOptionNames(): \Generator
     {
-        yield 'redis_sentinel';
-        yield 'sentinel_master';
+        yield ['redis_sentinel'];
+
```
Comment (xabbuh) [inline src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/RedisExtIntegrationTest.php]:
```suggestion
```

File: src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php (hunk 2 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php:50]: Yes, otherwise this condition will throw:

https://github.com/symfony/symfony/blob/49d00f66d5443ffd3b561bc43a4cc399d9716e4a/src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php#L232

File: src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/RedisExtIntegrationTest.php (hunk 3 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/RedisExtIntegrationTest.php]: Alright 👍

**Review by xabbuh — APPROVED** [review]

**Comment by xabbuh** [comment]
Good catch, thanks @alexandre-daubois.
